Patch from https://patch-diff.githubusercontent.com/raw/Tencent/rapidjson/pull/719.patch
to prevent build failure on gcc-14.2.0:
document.h:319:82: error: assignment of read-only member 'rapidjson::GenericStringRef<Char
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }

--- qt-everywhere-src-5.15.16/qtlocation/src/3rdparty/mapbox-gl-native/deps/rapidjson/1.1.0/include/rapidjson/document.h.orig	2024-11-08 14:36:17.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtlocation/src/3rdparty/mapbox-gl-native/deps/rapidjson/1.1.0/include/rapidjson/document.h	2025-01-13 22:02:40.296590177 +0100
@@ -316,8 +316,6 @@
 
     GenericStringRef(const GenericStringRef& rhs) : s(rhs.s), length(rhs.length) {}
 
-    GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
-
     //! implicit conversion to plain CharType pointer
     operator const Ch *() const { return s; }
 
@@ -328,6 +326,8 @@
     //! Disallow construction from non-const array
     template<SizeType N>
     GenericStringRef(CharType (&str)[N]) /* = delete */;
+     //! Copy assignment operator not permitted - immutable type
+    GenericStringRef& operator=(const GenericStringRef& rhs) /* = delete */;
 };
 
 //! Mark a character pointer as constant string
